pkg/app/pipectl/cmd/appconfig/migratefromdeployconfig.go

```diff
@@ -0,0 +1,183 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package appconfig
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/pipe-cd/pipecd/pkg/app/server/service/apiservice"
+	"github.com/pipe-cd/pipecd/pkg/cli"
+	"github.com/pipe-cd/pipecd/pkg/model"
+)
+
+type migrateFromDeployConfig struct {
+	root *command
+
+	repoRootPath string
+	repoID       string
+	envName      string
+	stdout       io.Writer
+}
+
+func newListCommand(root *command) *cobra.Command {
+	m := &migrateFromDeployConfig{
+		root:   root,
+		stdout: os.Stdout,
+	}
+	cmd := &cobra.Command{
+		Use:   "migrate-from-deploy-config",
+		Short: "Migrate local deployment configuration files to application configuration files. Basically this finds all deployment configuration files to add name field and env label to them.",
+		RunE:  cli.WithContext(m.run),
+	}
+
+	cmd.Flags().StringVar(&m.repoRootPath, "repo-root-path", m.repoRootPath, "The absolute path to the root directory of the Git repository.")
+	cmd.Flags().StringVar(&m.repoID, "repo-id", m.repoID, "The repository ID that is being registered in Piped config.")
+	cmd.Flags().StringVar(&m.envName, "env-name", m.envName, "The environment name.")
+
+	cmd.MarkFlagRequired("repo-root-path")
+	cmd.MarkFlagRequired("repo-id")
+	cmd.MarkFlagRequired("env-name")
+
+	return cmd
+}
+
+func (m *migrateFromDeployConfig) run(ctx context.Context, _ cli.Input) error {
+	cli, err := m.root.clientOptions.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to initialize client: %w", err)
+	}
+	defer cli.Close()
+
+	fmt.Fprintln(m.stdout, "Start finding and migrating deployment configuration files to application configuration files...")
+
+	var cursor string
+	var count int
+	for {
+		req := &apiservice.ListApplicationsRequest{
+			EnvName: m.envName,
+			Cursor:  cursor,
+		}
+		resp, err := cli.ListApplications(ctx, req)
+		if err != nil {
+			return fmt.Errorf("failed to list application: %w", err)
+		}
+
+		for _, app := range resp.Applications {
+			if app.GitPath.Repo.Id != m.repoID {
+				continue
+			}
+			if err := m.migrate(ctx, app); err != nil {
+				return err
+			}
+			count++
+		}
+		if resp.Cursor == "" {
+			break
+		}
+		cursor = resp.Cursor
+	}
+
+	fmt.Fprintf(m.stdout, "Successfully migrated %d applications\n", count)
+	return nil
+}
+
+func (m *migrateFromDeployConfig) migrate(app *model.Application) error {
+	configFilePath := filepath.Join(m.repoRootPath, app.GitPath.GetApplicationConfigFilePath())
+	oriData, err := os.ReadFile(configFilePath)
+	if err != nil {
+		return err
+	}
+
+	newData, err := convert(oriData, app.Name, m.envName, app.Description)
+	if err != nil {
+		return err
+	}
+
+	info, err := os.Stat(configFilePath)
+	if err != nil {
+		return err
+	}
+	return os.WriteFile(configFilePath, newData, info.Mode())
```

Comment:
Instead of writing to the disc one at a time, how about saving all `newData` in a map with `configFilePath` as a key, and writing to the file in parallel at the end?
This can
- prevent abnormal termination of the program with the file halfway modified due to errors caused by communication failures etc
- run I/O in parallel

Reply:
Nice idea. The first reason is worth doing.
Let me apply it.

Reply:
Applied!